sdk/core/azure-core/inc/azure/core/http/pipeline.hpp

```diff
@@ -52,13 +59,25 @@ namespace Azure { namespace Core { namespace Http {
      *
      * @param policies A sequence of #HttpPolicy representing a stack, first element corresponding
      * to the top of the stack.
+     *
+     * @throw `std::invalid_argument` when policies is empty.
      */
     explicit HttpPipeline(std::vector<std::unique_ptr<HttpPolicy>>&& policies)
```

Comment:
Unrelated, but why do we have this method that takes `vector<unique_ptr<HttpPolicy>>&&`?

Reply:
that's a move constructor.
see: https://en.cppreference.com/w/cpp/language/move_constructor

It enables creating an HttpPipeline from a temporal created policies like
`HttpPipeline(std::vector<...>(2))`  // The arg is a temporal value that is created to init the pipeline,  the `&&` enables us to use the temporal variable without creating a copy from it or trying to get a reference from it's adress (&).